ITISFoundation/osparc-simcore
# 🐛 Fixes registration in multiple products via invitations

## What do these changes do?

- 🐛A registered user gets an invitation to register in another product of the same deploy. This PR fixes a check upon registration that would not allow it
- 🐛 Fixes invitation guest email check to non-sensitive ( @eofli found a case of a user that was invited with upper-case email)
- ♻️some refactoring of login plugin (our plan is to split this further in coming PRs)

## Related issue/s

- Fixes on top of https://github.com/ITISFoundation/osparc-simcore/pull/4739

## How to test

- Driving test ``pytest -vv -k test_registration_to_different_product --pdb tests/unit``
- Manual exploratory testing (FYI @elisabettai 🚨)
   - Use workflow 1: register same user in two products
      - a registered user in product A
      - create an invitation for product B
      - gets added to product B
   - User workflow 2: fail to register same user in same product
      - a registered user in product A
      - create an invitation for product A
      - fails with invitation is already used  
   - User workflow 3: fail to register BANNED user
    - a registered user in product A is set to BANNED
    - create invitation for product C
    - should fail

## Dev Checklist


## DevOps Checklist
